Update base ubuntu version to 22.04

sayboras opened the pull request:
### Description

Please refer to individual commit for more details.

Relates: https://github.com/cilium/cilium/pull/20943

### Note

As per discussion offline, we can drop 4.9 build. Once this PR is merged, we will need to update Jenkins configuration.

### Testing

Testing was done with the below changes:
- Publish packer boxes with temp commit https://github.com/cilium/packer-ci-build/pull/359
- Cilium with new boxes https://github.com/cilium/cilium/pull/24565

pchaigno committed:
packer: Update to Ubuntu 22.04

Relates: https://github.com/cilium/cilium/pull/20943

Signed-off-by: Paul Chaignon <paul@cilium.io>
Signed-off-by: Tam Mach <tam.mach@cilium.io>

pchaigno committed:
packer: Update boot commands for Ubuntu 22.04

The old approach of passing the cloud-init config doesn't work anymore.
I found the new one from a random GitHub repository [1]. Yay GitHub!

1 - https://github.com/simula/nornet-vmimage-builder-scripts/blob/ce6490e94e040f3bf0304d9dd11da4abb2c1aa71/installer/ubuntu-22.04.json
Signed-off-by: Paul Chaignon <paul@cilium.io>

pchaigno committed:
kernel: Install libssl1.1 for kernel downgrade

An older version of libssl is needed to install older kernels.

Co-authored-by: Tam Mach <tam.mach@cilium.io>
Signed-off-by: Paul Chaignon <paul@cilium.io>

sayboras committed:
packer: Add wait step for cloud-init completion

Signed-off-by: Tam Mach <tam.mach@cilium.io>

sayboras committed:
dep: Use forked netperf to fix compilation issue

We can just move it to cilium org if required :).

Signed-off-by: Tam Mach <tam.mach@cilium.io>

sayboras committed:
dep: Update cri-o version

This commit is to bump OS version and cri-o version to recent one.

Signed-off-by: Tam Mach <tam.mach@cilium.io>

sayboras committed:
dep: Update packages for 22.04

This commit is to make sure additional package installation will not
fail due to not found error.

Signed-off-by: Tam Mach <tam.mach@cilium.com>

sayboras committed:
packer: Fix crash during cloud-config installation

```
/usr/bin/ckbcomp: Can not find file "symbols/en" in any known directory
```

Signed-off-by: Tam Mach <tam.mach@cilium.io>

sayboras committed:
kernel/5.4: Update linux modules path for 5.4

Signed-off-by: Tam Mach <tam.mach@cilium.io>

sayboras committed:
kernel: Update COMPRESS algo in initramfs.com

This is to avoid kernel panic due to the below issue during booting up.

```
VFS: Cannot open root device "UUID=..." or unknown-block(0,0): error -6
Please append a correct "root=" boot option; here are the available partitions:
Kernel panic - not syncing: VFS: Unable to mount root fs on unknown-block(0,0)
...
```

Suggested-by: Maxim Mikityanskiy <maxim@isovalent.com>
Signed-off-by: Tam Mach <tam.mach@cilium.io>

sayboras committed:
kernel: Add execute permission for kernel-next-tools.sh

Mainly for local development, as the CI is working as it was.

Signed-off-by: Tam Mach <tam.mach@cilium.io>

sayboras commented:
/build

sayboras commented:
/build

nbusseneau commented:
🙏🏻

pchaigno commented:
I think you're missing a few :pray: to make it work. 2 CI jobs are passing so you probably want to double the number of :pray: to get all CI jobs.

sayboras committed:
docker: Update the docker installation steps

This is to follow to the latest step mentioned in the official docs.

https://docs.docker.com/engine/install/ubuntu/

Signed-off-by: Tam Mach <tam.mach@cilium.io>

sayboras committed:
docker: Set alternatives for iptables

This is to avoid the below issue while starting up docker.

```
Mar 24 22:59:10 ubuntu-jammy dockerd[2820]: failed to start daemon: Error initializing network controller: error obtaining controller instance: failed to create NAT chain DOCKER: iptables failed: iptables -t nat -N DOCKER: iptables v1.8.7 (nf_tables): Could not fetch rule set generation id: Invalid argument
```

Relates: https://github.com/docker/for-linux/issues/1437

Signed-off-by: Tam Mach <tam.mach@cilium.io>

sayboras committed:
kernel: Remove support for kernel 4.9

It wasn't easy for us to do, old friend. You're serving us well, but we
are transferring to 4.19 now. I'm afraid you can't come :(.

Signed-off-by: Tam Mach <tam.mach@cilium.io>

sayboras commented:
/build

sayboras commented:
/build

sayboras commented:
/build

sayboras commented:
/build-4.19

sayboras committed:
Temp commit for testing purpose

sayboras commented:
/build

sayboras committed:
kernel: Disable cgroupv2 for 4.19

This is to support k8s-1.16-kernel-4.19 job.

sayboras commented:
/build-4.19

sayboras commented:
/build-4.19

sayboras commented:
/build-net-next

sayboras commented:
/build-5.4

ldelossa reviewed (approved):
Approved, contingent on @gentoo-root's suggestions getting addressed.

sayboras commented:
> Are you planning to continue working on this?

We are moving to another little vm helper based tests in main, the only value we will have is for v1.13.

nbusseneau commented:
@gentoo-root I think @sayboras is referring to the new Ginkgo workflow: https://github.com/cilium/cilium/blob/main/.github/workflows/conformance-ginkgo.yaml
We don't use Vagrant for this one, so I believe the issue referenced becomes obsolete for `main`. However the Ginkgo workflow is not currently planned to be backported, so the issue will still be there for stable branches.

tzssangglass commented:
wait for this PR, As a beginner in Cilium, I encountered an issue with mismatched versions of glibc and clang when following [using-the-provided-vagrantfiles-recommended](https://github.com/cilium/cilium/blob/main/Documentation/contributing/development/dev_setup.rst#option-1---using-the-provided-vagrantfiles-recommended).

I think this is due to the low version of glibc in vagrant `cilium/ubuntu-4-19`.
